modules/ztp-image-based-upgrade-with-backup.adoc

```diff
@@ -0,0 +1,229 @@
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "3"
+spec:
+  backupName:
+    apache-app
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: localvolume
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    localvolume
+----
+
+. Create a `ConfigMap` CR for the backup and restore information that the {lcao} uses during the upgrade.
+
+.. Create a `kustomization.yaml` that appends the information to a `ConfigMap` CR:
++
+[source,yaml]
+----
+configMapGenerator:
+- name: oadp-cm
+  namespace: openshift-adp
+  files:
+  - backup_acm_klusterlet.yaml
+  - backup_localvolume.yaml
+  - backup_app.yaml
+  - restore_acm_klusterlet.yaml
+  - restore_localvolumes.yaml
+  - restore_app.yaml
+----
+
+.. Create the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ kustomize build ./ -o OadpCm.yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-7gtgdhmdc5
+  namespace: openshift-adp
+[...]
+----
+
+.. Apply the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ oc apply -f OadpCm.yaml
+----
+
+. Append the `imageBasedUpgrade` CR with the backup and restore information:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: upgrade
+spec:
+  stage: Idle #TODO might need to change or move these steps
+  seedImageRef:
+    version: 4.14.6
+    image: <seed_container_image>
+  oadpContent:
+  - name: oadp-cm-7gtgdhmdc5
+    namespace: openshift-adp
+----
+
+The OADP Operator creates a backup of the data specified in the OADP `Backup` and `Restore` CRs.
+//TODO provide examples with the OADP CRs
+
+. Move to the upgrade stage by changing the `Prep` value to `Upgrade` in the `ImageBasedUpgrade` CR.
++
+[source,terminal]
+----
+$ oc patch imagebasedupgrades.lca.openshift.io upgrade -p='{"spec": {"stage": "Upgrade"}}' --type=merge
+----
+
+. Check the status of the `imageBasedUpgrade` CR:
++
+[source,terminal]
+----
+$ oc get imagebasedupgrades.lca.openshift.io upgrade -oyaml
+----
+
++
+.Example output
+[source,yaml]
+----
+status:
+  conditions:
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: In progress
+    observedGeneration: 2
+    reason: InProgress
+    status: "False"
+    type: Idle
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: 'Prep completed: total: 121 (pulled: 1, skipped: 120, failed: 0)'
+    observedGeneration: 2
+    reason: Completed
+    status: "True"
+    type: PrepCompleted
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: Prep completed
+    observedGeneration: 2
+    reason: Completed
+    status: "False"
+    type: PrepInProgress
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: Upgrade completed
+    observedGeneration: 3
+    reason: Completed
+    status: "True"
+    type: UpgradeCompleted
+----
+
+. The target cluster reboots.
```

Comment:
Before rebooting, the backups are executed and stored in the S3 bucket we configured in the DataProtectionApplication. This information can be monitored from the logs of LCA, I mean, if the backup was successful or not. 

Also it can troubleshoot by checking the S3 compatible storage solution, check if the info is there or not. Or by executing in the target cluster:

~~~sh
$ oc get backups -n openshift-adp -o custom-columns=NAME:.metadata.name,Status:.status.phase,Reason:.status.failureReason
NAME             Status      Reason
acm-klusterlet   Completed   <none>
apache-app       Completed   <none>
localvolume      Completed   <none>
~~~